File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/ReviewsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ReviewsRow } from "./ReviewsRow";

export interface ReviewsColumns {
    Id: Column<ReviewsRow>;
    Name: Column<ReviewsRow>;
    Email: Column<ReviewsRow>;
    Subject: Column<ReviewsRow>;
    Description: Column<ReviewsRow>;
}

export class ReviewsColumns extends ColumnsBase<ReviewsRow> {
    static readonly columnsKey = 'ECommerceWebsite.Reviews';
    static readonly Fields = fieldsProxy<ReviewsColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Reviews/ReviewsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.Reviews")]
[BasedOnRow(typeof(ReviewsRow), CheckNames = true)]
public class ReviewsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Email { get; set; }
    public string Subject { get; set; }
    public string Description { get; set; }
}